src/pip/_internal/req/req_install.py

```diff
@@ -898,32 +900,34 @@ def archive(self, build_dir):
                 shutil.move(archive_path, dest_file)
             elif response == 'a':
                 sys.exit(-1)
-        if create_archive:
-            zip = zipfile.ZipFile(
-                archive_path, 'w', zipfile.ZIP_DEFLATED,
-                allowZip64=True
-            )
+
+        if not create_archive:
+            return
+
+        with zipfile.ZipFile(
+            archive_path, 'w', zipfile.ZIP_DEFLATED, allowZip64=True
+        ) as zip:
             dir = os.path.normcase(os.path.abspath(self.setup_py_dir))
             for dirpath, dirnames, filenames in os.walk(dir):
                 if 'pip-egg-info' in dirnames:
                     dirnames.remove('pip-egg-info')
                 for dirname in dirnames:
-                    dir_arcname = self._get_archive_name(dirname,
-                                                         parentdir=dirpath,
-                                                         rootdir=dir)
+                    dir_arcname = self._get_archive_name(
+                        dirname, parentdir=dirpath, rootdir=dir
```

Comment:
```suggestion
                        dirname, parentdir=dirpath, rootdir=dir,
```

Reply:
I thought we were vaguely following the Black style, which omits trailing comma when args are on a single line. I looked in the code base and it seems like no trailing comma is a bit more common (but there are probably some false positives):

```
$ pwd
/tmp/pip/src/pip/_internal
$ grep -A 2 -r '($' | grep -B 2 ')$' | grep -v -e ')$' -e '($' -e '--' | grep -c '[,]$'
67
$ grep -A 2 -r '($' | grep -B 2 ')$' | grep -v -e ')$' -e '($' -e '--' | grep -c '[^,]$'
103
```

Reply:
The point here is that if we add an argument later, the previous line shouldn't need modification -- it's an improvement we've been making slowly as we work on parts of the code.

definitely OT: when black puts arguments on the same line, the entire call (name + parens + arguments) are on that line. And when it's breaking the call, each argument gets its own line. :)

Reply:
Continuing OT: the style guide indicates that there's an intermediate step where it tries to break on the outermost matching bracket/parenthesis ([ref](https://black.readthedocs.io/en/stable/the_black_code_style.html#how-black-wraps-lines))

We don't follow the one-line-per-argument universally though, so I agree with the benefit here and will add the trailing comma. :+1:

Reply:
Done [here](https://github.com/pypa/pip/pull/6988/files#diff-47a4ece629e4698a5eaefea075d3cbdfR917).